controllers/hostingdeploymentautoscalingjob/hostingdeploymentautoscalingjob_controller.go

```diff
@@ -0,0 +1,448 @@
+/*
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostingdeploymentautoscalingjob
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/amazon-sagemaker-operator-for-k8s/controllers"
+	"github.com/aws/amazon-sagemaker-operator-for-k8s/controllers/sdkutil"
+	"github.com/aws/amazon-sagemaker-operator-for-k8s/controllers/sdkutil/clientwrapper"
+	aws "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/applicationautoscaling"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	hostingdeploymentautoscalingjobv1 "github.com/aws/amazon-sagemaker-operator-for-k8s/api/v1/hostingdeploymentautoscalingjob"
+)
+
+// All the status used by the controller during reconciliation.
+// This operator includes two steps. For the rest of the file these are - Step1: RegisterTargets; Step2: PutScalingPolicy
+const (
+	// The process of creation has started and is in-progress.
+	ReconcilingAutoscalingJobStatus = "ReconcilingAutoscalingJob"
+
+	// This Status signifies that the job has been successfully completed for both steps
+	CreatedAutoscalingJobStatus = "CreatedAutoscalingJob"
+
+	// Could have failed either at step1 or step2
+	FailedAutoscalingJobStatus = "FailedAutoscalingJob"
```

Comment:
Does this represent an error state? Typically when a pod fails to create it will be in `Error`.

Reply:
ok, I changed the statuses to error, created, reconciling. Still not the best but couldn't think of anything else.